lib/login_screen.dart

```diff
@@ -0,0 +1,268 @@
+              ),
+            ),
+            child: SizedBox(
+              width: double.infinity,
+              height: 60,
+              child: ElevatedButton(
+                onPressed: allPassed
+                    ? () async {
+                        _login(
+                          emailController.text,
+                          passwordController.text,
+                        );
+                      }
+                    : null,
+                child: const Text(
+                  "Login",
+                  style: TextStyle(
+                    fontWeight: FontWeight.w600,
+                    fontSize: 16,
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  _login(email, password) {
+    // Uses the API library to login
+    api.login(email, password).then((response) {
+      Map<String, dynamic> parsed = jsonDecode(response.body);
+
+      if (kDebugMode) {
+        debugPrint(response.body);
+      }
+
+      // If the response contains an access token, the login was successful
+      if (parsed['access_token'] != null) {
+        loginFail = false;
+
+        if (kDebugMode) {
+          debugPrint("true");
+        }
+        // Uses API library to store token
+        api.storeToken(parsed['access_token']);
+
+        // Uses the API library to store the login info if the user wants to automatically login
+        if (autologin) {
+          api.storeLoginInfo(email, password);
+        }
+
+        // unfocus the textfield
+        FocusScope.of(context).unfocus();
+
+        // Navigate to the home screen
+        Navigator.pushNamed(context, '/home');
+      } else {
+        // Clear password, fail login and display error message
+        passwordController.clear();
+        loginFail = true;
+        errorMessage = parsed['message'];
+
+        // unfocus the textfield
+        FocusScope.of(context).unfocus();
+      }
+    }).onError((error, stackTrace) {
+      // Clear password, fail login and display error message
+      passwordController.clear();
+      loginFail = true;
+      errorMessage = "Login information is invalid";
+
+      // unfocus the textfield
+      FocusScope.of(context).unfocus();
+    });
+  }
+
+  void autoLoginCounter() {
+    // Every hour check if logged in and then refresh bearer token
+    Timer.periodic(const Duration(seconds: 3600), (timer) async {
+      setState(() {
+        autoLoginCheck();
+      });
+    });
+  }
+
+  autoLoginCheck() async {
+    // Check if the users data is stored and if so, log them in
+    if (await api.isLoggedIn(autoLogin: true)) {
+      // Logs the user in
+      if (await api.autoLogin()) {
+        if (mounted) {
+          // Start the auto login refresh token timer
+          autoLoginCounter();
+          // Navigate to the home screen
+          Navigator.pushNamed(context, '/home');
+        }
+      }
+    }
+  }
```

Comment:
Keeping the login information indefinitely seems unsafe and should probably be limited to 30 days or less similarly to Postman.

Reply:
We are no longer storing the login info only the tokens. 
They will be deleted on logout.